pkg/packetfilter/nftables/nftables.go

```diff
@@ -0,0 +1,511 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nftables
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/admiral/pkg/log"
+	"github.com/submariner-io/submariner/pkg/packetfilter"
+	"golang.org/x/exp/slices"
+	"k8s.io/utils/ptr"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/knftables"
+)
+
+const (
+	/* Single table named 'submariner' is used for nftables configuration.*/
+	submarinerTable = "submariner"
+)
+
+var (
+	iphookChainTypeToNftablesType = map[packetfilter.ChainType]knftables.BaseChainType{
+		packetfilter.ChainTypeFilter: knftables.FilterType,
+		packetfilter.ChainTypeRoute:  knftables.RouteType,
+		packetfilter.ChainTypeNAT:    knftables.NATType,
+	}
+
+	iphookChainHookToNftablesHook = map[packetfilter.ChainHook]knftables.BaseChainHook{
+		packetfilter.ChainHookPrerouting:  knftables.PreroutingHook,
+		packetfilter.ChainHookInput:       knftables.InputHook,
+		packetfilter.ChainHookForward:     knftables.ForwardHook,
+		packetfilter.ChainHookOutput:      knftables.OutputHook,
+		packetfilter.ChainHookPostrouting: knftables.PostroutingHook,
+	}
+
+	iphookChainTypeToNftablesBasicPriority = map[packetfilter.ChainType]knftables.BaseChainPriority{
+		packetfilter.ChainTypeFilter: knftables.FilterPriority,
+		packetfilter.ChainTypeRoute:  knftables.ManglePriority,
+		packetfilter.ChainTypeNAT:    knftables.DNATPriority,
+	}
+
+	ruleActionToStr = map[packetfilter.RuleAction][]string{
+		packetfilter.RuleActionAccept: {"accept"},
+		packetfilter.RuleActionMss:    {"tcp", "option", "maxseg"},
+		packetfilter.RuleActionMark:   {"meta", "mark"},
+		packetfilter.RuleActionSNAT:   {"snat"},
+		packetfilter.RuleActionDNAT:   {"dnat"},
+		packetfilter.RuleActionJump:   {"jump"},
+	}
+
+	logger = log.Logger{Logger: logf.Log.WithName("NFTables")}
+)
+
+type RuleSpec []string
+
+func (r RuleSpec) String() string {
+	return strings.Join(r, " ")
+}
+
+type packetFilter struct {
+	nftables knftables.Interface
+}
+
+func New() (packetfilter.Driver, error) {
+	nft, err := knftables.New(knftables.IPv4Family, submarinerTable)
```

Comment:
`namedset.go` supports IPv6 (at least, appears to), but this hard-codes IPv4; is that intentional?

Reply:
Yep, we have the same issue also with iptables , [iptables hardcoded to v4](https://github.com/submariner-io/submariner/blob/devel/pkg/packetfilter/iptables/iptables.go#L77) while iptables/namedset supports also [v6](https://github.com/submariner-io/submariner/blob/devel/pkg/packetfilter/iptables/namedset.go#L33)  .

I'll delete https://github.com/submariner-io/submariner/blob/devel/pkg/packetfilter/packetfilter.go#L257 and update the relevant code.